src/garage/torch/policies/context_conditioned_policy.py

```diff
@@ -0,0 +1,268 @@
+# pylint: disable=attribute-defined-outside-init, protected-access
+"""A policy used in training meta reinforcement learning algorithms.
+
+It is used in PEARL (Probabilistic Embeddings for Actor-Critic Reinforcement
+Learning). The paper on PEARL can be found at https://arxiv.org/abs/1903.08254.
+Code is adapted from https://github.com/katerakelly/oyster.
+"""
+
+import numpy as np
+import torch
+from torch import nn
+import torch.nn.functional as F
+
+import garage.torch.utils as tu
+
+
+class ContextConditionedPolicy(nn.Module):
+    """A policy that outputs actions based on observation and latent context.
+
+    In PEARL, policies are conditioned on current state and a latent context
+    (adaptation data) variable Z. This inference network estimates the
+    posterior probability of z given past transitions. It uses context
+    information stored in the encoder to infer the probabilistic value of z and
+    samples from a policy conditioned on z.
+
+    Args:
+        latent_dim (int): Latent context variable dimension.
+        context_encoder (garage.torch.embeddings): Recurrent or
+            permutation-invariant context encoder.
+        policy (garage.torch.policies.Policy): Policy used to train the
+            network.
+        use_ib (bool): True if latent context is not deterministic; false
+            otherwise
+        use_next_obs (bool): True if next observation is used in context
+            for distinguishing tasks; false otherwise.
+
+    """
+
+    def __init__(self, latent_dim, context_encoder, policy, use_ib,
+                 use_next_obs):
+        super().__init__()
+        self._latent_dim = latent_dim
+        self._context_encoder = context_encoder
+        self._policy = policy
+        self._use_ib = use_ib
+        self._use_next_obs = use_next_obs
+
+        # initialize buffers for z distribution and z
+        # use buffers so latent context can be saved along with model weights
+        # z_means and z_vars are the params for the gaussian distribution
+        # over latent task belief maintained in the policy; z is a sample from
+        # this distribution that the policy is conditioned on
+        self.register_buffer('z', torch.zeros(1, latent_dim))
+        self.register_buffer('z_means', torch.zeros(1, latent_dim))
+        self.register_buffer('z_vars', torch.zeros(1, latent_dim))
+
+        self.reset_belief()
+
+    def reset_belief(self, num_tasks=1):
+        """Reset q(z|c) to the prior and sample a new z from the prior.
```

Comment:
you can use math environments to make these equations render as LaTeX. see `src/garage/_dtypes.py` for an example.